test/integ/cli/test_cli_attachment.py

```diff
@@ -0,0 +1,285 @@
+                "s3://cross-account-bucket/test",
+            ],
+        )
+        assert (
+            result.exit_code != 0
+        ), f"Expecting cross-account s3 access to fail but not, CLI output {result.output}"
+        assert "deadline.job_attachments.exceptions.JobAttachmentsS3ClientError" in result.output
+        assert "Error uploading file" in result.output
+
+        result = runner.invoke(
+            main,
+            [
+                "attachment",
+                "download",
+                "--manifests",
+                manifest_path,
+                "--profile",
+                "default",
+                "--s3-root-uri",
+                "s3://cross-account-bucket/test",
+            ],
+        )
+        assert (
+            result.exit_code != 0
+        ), f"Expecting cross-account s3 access to fail but not, CLI output {result.output}"
+        assert "deadline.job_attachments.exceptions.JobAttachmentsS3ClientError" in result.output
+        assert "Error downloading file" in result.output
+
+    @pytest.mark.parametrize("manifest_case_key", MOCK_MANIFEST_CASE.keys())
+    def test_attachment_basic_flow(self, temp_dir, job_attachment_resources, manifest_case_key):
+
+        # Given
+        file_name: str = f"{hash_data(temp_dir.encode('utf-8'), HashAlgorithm.XXH128)}_output"
+        manifest_path: str = os.path.join(temp_dir, file_name)
+
+        with open(
+            manifest_path,
+            "w",
+            encoding="utf8",
+        ) as f:
+            json.dump(MOCK_MANIFEST_CASE[manifest_case_key], f)
+
+        s3_root_uri = f"s3://{job_attachment_resources.bucket_name}/{job_attachment_resources.bucket_root_prefix}"
+
+        runner = CliRunner()
+        # Temporary, always add cli_attachment until launched.
+        main.add_command(cli_attachment)
+
+        # When - test upload the local asset file
+        result = runner.invoke(
+            main,
+            [
+                "attachment",
+                "upload",
+                "--manifests",
+                manifest_path,
+                "--root-dirs",
+                temp_dir,
+                "--profile",
+                "default",
+                "--s3-root-uri",
+                s3_root_uri,
+            ],
+        )
+        # Then
+        assert result.exit_code == 0, f"Non-Zeo exit code, CLI output {result.output}"
+
+        # When - test download the file just uploaded
+        result = runner.invoke(
+            main,
+            [
+                "attachment",
+                "download",
+                "--manifests",
+                manifest_path,
+                "--profile",
+                "default",
+                "--s3-root-uri",
+                s3_root_uri,
+                "--json",
+            ],
+        )
+        # Then
+        assert result.exit_code == 0, f"Non-Zeo exit code, CLI output {result.output}"
+        assert json.loads(result.output)["processed_bytes"] == TEST_FILE_SIZE
```

Comment:
nit, should we check the content of what is downloaded?

Reply:
Added check for file name downloaded, and updated the check on processed bytes.